scripts/build-android.sh

```diff
@@ -3,16 +3,17 @@
 # Needed to fail on must_get_env()
 set -e
 
-GIT_ROOT=$(cd "${BASH_SOURCE%/*}" && git rev-parse --show-toplevel)
+GIT_ROOT=$(cd "$(dirname "${BASH_SOURCE[0]}")" && git rev-parse --show-toplevel)
 source "${GIT_ROOT}/scripts/colors.sh"
 
 function must_get_env() {
-  declare -n VAR_VALUE="$1"
+  local VAR_NAME="$1"
+  declare -n VAR_VALUE="$VAR_NAME"
   if [[ -n "${VAR_VALUE}" ]]; then
       echo "${VAR_VALUE}"
       return
   fi
-  echo -e "${RED}No required env variable:${RST} ${BLD}${!VAR_VALUE}${RST}" 1>&2
+  echo -e "${RED}No required env variable:${RST} ${BLD}${VAR_NAME}${RST}" 1>&2
```

Comment:
The `!` trick works because `VAR_VALUE` is not a variable but rather a reference to a var created via `declare -n`:
```
      -n	make NAME a reference to the variable named by its value
```
Which means that `VAR_VALUE` is a reference to `$1`. We can test this:
```bash
 > TEST=something
 > VAR_NAME=TEST
 > declare -n VAR_REF=$VAR_NAME
 > echo $VAR_REF
something
 > echo ${!VAR_REF}
TEST
```
Your version is clearer, but the result is the same. What I would like to understand is why for you it doesn't expand into the name of variable from the `nameref`.